velox/expression/FieldReference.h

```diff
@@ -33,6 +33,19 @@ class FieldReference : public SpecialForm {
             false /* trackCpuUsage */),
         field_(field) {}
 
+  FieldReference(
+      TypePtr type,
+      const std::vector<ExprPtr>& inputs,
+      int32_t index)
+      : SpecialForm(
+            std::move(type),
+            inputs,
+            inputs.at(0)->type()->asRow().nameOf(index),
+            false, /* supportsFlatNoNullsFastPath */
+            false /* trackCpuUsage */),
+        field_(inputs.at(0)->type()->asRow().nameOf(index)),
+        index_(index) {}
+
   const std::string& field() const {
     return field_;
```

Comment:
Can we add a non-empty check here as well?

Reply:
Why?  Again, that feels specific to the requirements of the caller.  If you know of a place that requires that I'm happy to add a check there.

Reply:
Can we add a check in the constructor of `Subfield::NestedField` that the field name is not empty?